dev/tools/bin/engine_hash.sh

```diff
@@ -5,70 +5,27 @@
 
 # ---------------------------------- NOTE ---------------------------------- #
 #
-# We must keep the logic in this file consistent with the logic in the
-# `engine_hash.dart` script in the same directory to ensure that Flutter
-# continues to work across all platforms!
+# This file will appear unused within the monorepo. It is used internally
+# (in google3) as part of the roll process, and care should be put before 
+# making changes.
+#
+# See cl/688973229.
 #
 # -------------------------------------------------------------------------- #
 
-# TODO(codefu): Add a test that this always outputs the same hash as
-# `engine_hash.dart` when the repositories are merged
-
-STRATEGY=head
+# Test for fusion repository
+if [ -f "$FLUTTER_ROOT/DEPS" ] && [ -f "$FLUTTER_ROOT/engine/src/.gn" ]; then
```

Comment:
Is FLUTTER_ROOT set already?

Reply:
Simplified further, and fixed, and tested :)